at_secondary/at_secondary_server/test/resource_manager_test.dart

```diff
@@ -0,0 +1,158 @@
+import 'dart:io';
+
+import 'package:at_persistence_secondary_server/at_persistence_secondary_server.dart';
+import 'package:at_secondary/src/connection/outbound/outbound_client.dart';
+import 'package:at_secondary/src/notification/at_notification_map.dart';
+import 'package:at_secondary/src/notification/queue_manager.dart';
+import 'package:at_secondary/src/notification/resource_manager.dart';
+import 'package:at_secondary/src/server/at_secondary_impl.dart';
+import 'package:test/test.dart';
+import 'package:mocktail/mocktail.dart';
+
+class MockOutboundClient extends Mock implements OutboundClient {}
+
+void main() async {
+  OutboundClient mockOutboundClient = MockOutboundClient();
+  ResourceManager rm = ResourceManager.getInstance();
+  var storageDir = Directory.current.path + '/test/hive';
+
+  setUp(() {
+    reset(mockOutboundClient);
+    when(() => mockOutboundClient.notify(any())).thenAnswer((_) async {
+      throw Exception('Failed to Notify');
+    });
+  });
+
+  group('A group of notify verb test', () {
+    setUp(() async => await setUpFunc(storageDir));
+    test('Test send notifications', () async {
+      var atNotification1 = (AtNotificationBuilder()
+            ..id = '121'
+            ..atValue = 'bob@gmail.com'
+            ..notification = 'email'
+            ..fromAtSign = '@bob'
+            ..toAtSign = '@alice'
+            ..strategy = 'all'
+            ..opType = OperationType.update
+            ..ttl = -1
+            ..retryCount = 1)
+          .build();
+      var atNotification2 = (AtNotificationBuilder()
+            ..id = '122'
+            ..atValue = '90192019021'
+            ..fromAtSign = '@bob'
+            ..toAtSign = '@alice'
+            ..strategy = 'all'
+            ..opType = OperationType.update
+            ..notification = 'phone'
+            ..retryCount = 1)
+          .build();
+      var atNotification3 = (AtNotificationBuilder()
+            ..id = '123'
+            ..fromAtSign = '@bob'
+            ..toAtSign = '@alice'
+            ..atValue = 'USA'
+            ..strategy = 'all'
+            ..opType = OperationType.update
+            ..notification = 'location'
+            ..retryCount = 1)
+          .build();
+      var atsign = '@alice';
+      Iterator notificationIterator =
+          [atNotification1, atNotification2, atNotification3].iterator;
+      await rm.sendNotifications(
+          atsign, mockOutboundClient, notificationIterator);
+      var atNotificationList = [];
+      var itr = QueueManager.getInstance().dequeue(atsign);
+      while (itr.moveNext()) {
+        atNotificationList.add(itr.current);
+      }
+      expect(atNotificationList[0].id, '121');
+      expect(atNotificationList[1].id, '122');
+      expect(atNotificationList[2].id, '123');
```

Comment:
Please add a code comment with some explanation of the meaning of the test - i.e. what behaviour are we verifying, and how are we making it happen